packages/backend-output-storage/src/stack_metadata_output_storage_strategy.test.ts

```diff
@@ -49,4 +50,90 @@ void describe('StackMetadataBackendOutputStorageStrategy', () => {
       backendOutputStackMetadataSchema.parse(template.toJSON().Metadata);
     });
   });
+
+  void describe('appendToBackendOutputList', () => {
+    void it('adds list to stack output and metadata for entry', () => {
+      const app = new App();
+      const stack = new Stack(app);
+      const outputStorage = new StackMetadataBackendOutputStorageStrategy(
+        stack
+      );
+      outputStorage.appendToBackendOutputList('TestStorageOutput', {
+        version: '1',
+        payload: {
+          something: 'special',
+        },
+      });
+
+      const template = Template.fromStack(stack);
+      template.hasOutput('something', { Value: JSON.stringify(['special']) });
+      template.templateMatches({
+        Metadata: {
+          TestStorageOutput: {
+            version: '1',
+            stackOutputs: ['something'],
+          },
+        },
+      });
+    });
+
+    void it('appends to an existing list in stack output', () => {
+      const app = new App();
+      const stack = new Stack(app);
+      const outputStorage = new StackMetadataBackendOutputStorageStrategy(
+        stack
+      );
+      outputStorage.appendToBackendOutputList('TestStorageOutput', {
+        version: '1',
+        payload: {
+          something: 'special',
+        },
+      });
+      outputStorage.appendToBackendOutputList('TestStorageOutput', {
+        version: '1',
+        payload: {
+          something: 'otherSpecial',
+        },
+      });
+      const template = Template.fromStack(stack);
+      template.hasOutput('something', {
+        Value: JSON.stringify(['special', 'otherSpecial']),
+      });
+      template.templateMatches({
+        Metadata: {
+          TestStorageOutput: {
+            version: '1',
+            stackOutputs: ['something'],
+          },
+        },
+      });
+    });
+
+    void it('throws when trying to add to same list with different version', () => {
+      const app = new App();
+      const stack = new Stack(app);
+
+      const outputStorage = new StackMetadataBackendOutputStorageStrategy(
+        stack
+      );
+      stack.node.addMetadata('TestStorageOutput', {
+        version: '1',
+        stackOutputs: ['special'],
+      });
```

Comment:
rather than setting this directly, we should test that two calls to appendToBackendOutputList with different versions fails on the second call